controllers/storagecluster/cephcluster.go

```diff
@@ -528,8 +528,8 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	// Enable the msgr2 port always if it's not an external cluster
-	if !sc.Spec.ExternalStorage.Enable {
+	// Enable the msgr2 port always if it's not an provider or external/consumer cluster
```

Comment:
// Enable the msgr2 port always if it's not an provider or external/consumer cluster

//Don't enable msg2 port if it's the provider or external/consumer cluster